bson/primitive/decimal.go

```diff
@@ -114,6 +121,80 @@ Loop:
 	return string(repr[last+pos:])
 }
 
+// BigInt returns significand as big.Int and exponent, bi * 10 ^ exp.
+func (d Decimal128) BigInt() (bi *big.Int, exp int, err error) {
+	h, l := d.GetBytes()
+	var pos int // positive sign
+
+	if h>>63&1 == 0 {
+		pos = 1
+	}
+
+	switch h >> 58 & (1<<5 - 1) {
+	case 0x1F:
+		return nil, 0, fmt.Errorf("cannot parse NaN as a *big.Int")
+	case 0x1E:
+		return nil, 0, fmt.Errorf("cannot parse %s as a *big.Int", "-Infinity"[pos:])
+	}
+
+	if h>>61&3 == 3 {
+		// Bits: 1*sign 2*ignored 14*exponent 111*significand.
+		// Implicit 0b100 prefix in significand.
+		exp = int(h>>47&(1<<14-1)) + MinDecimal128Exp
```

Comment:
Ditto comment about avoiding the duplicated `MinDecimal128Exp`, I believe it is possible to do that here as well.